app/controllers/posts_controller.rb

```diff
@@ -8,6 +8,41 @@ def index
   end
 
   def show
+    @user = current_user
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+    @user = current_user
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = current_user
+    if @post.save
+      puts @post
+      redirect_to user_posts_path(current_user)
+    else
+      render :new
+    end
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    Like.new(author: current_user, post: @post).save
+    redirect_to user_posts_path(current_user)
+  end
+
+  def comment
+    @post = Post.find(params[:id])
+    p params
+    @comment = Comment.new(author: current_user, post: @post, text: params[:text])
+    @comment.save
+    redirect_to user_posts_path(current_user)
+  end
```

Comment:
### **Great** implementation of the project requirements.

- [ ] Kindly create a `CommentsController` action for post comments and refactor your code.

- A comment is created on behalf of the currently logged in user (current_user) of the app. The comments controller should have new and create actions.